src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -2228,6 +2232,16 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
     }
 
+    winrt::fire_and_forget ControlCore::_terminalMenuChanged(std::wstring_view menuJson,
+                                                             unsigned int replaceLength)
+    {
+        auto args = winrt::make_self<MenuChangedEventArgs>(winrt::hstring{ menuJson },
+                                                           replaceLength);
+
+        co_await winrt::resume_background();

```

Comment:
aren't we already on the background?

Reply:
Yes BUT we're on the _output_ thread. I did this to try and get us off the output thread entirely. I assumed (possibly wrongly) that `resume_background` would always go to a different thread.

Reply:
ah it might. good call.